### DISCUSSION TITLE
Solution: Simple Steps to Install 'Atri' in Windows Machine.

### DISCUSSION THREAD

**Discussion by chiragmathurid** [Atri-Labs/atrilabs-engine#749] [Help: Installation & Start]
I have noticed that many people encounter issues with the 'pipenv install atri' command. Prior to installing the pipenv tool, it is necessary to have Python and pip already installed on the computer.

Below are the simple steps that help to proceed. For smooth installation, you can delete the existing version of Python in your PC 
Reference: https://www.wikihow.com/Uninstall-Python

1.  Open Windows Powershell
2.  Check Python version: ***python -V*** (Capital V) [If Installed, then Make sure that you have Python>=3.8 &<=3.10.8]
3.  If not, then Install python v3.10: https://www.python.org/ftp/python/3.10.0/python-3.10.0-amd64.exe. 
     Before proceeding with installation, select Add python.exe to PATH 
4.  Now, Check the Python version again (Step-2). You will get:- _Python 3.10.0_
5.  Check if you have the pip tool on your computer: 
     Command:- ***pip -V***. 
     You will get:-  _pip 21.2.3 from C:\Program Files\Python310\lib\site- packages\pip (python 3.10)_

**Install pipenv on Windows.**

1. Run ***pip install pipenv*** command in Windows Powershell
2. Then, Go to This PC -> Properties -> Advanced System Settings -> **Advanced** tab 
3. Click on **Environment Variable**
4. Double Click on **Path** in User variables [above section]
5. Click on **New** and add the following:
   C:\Users\your username\AppData\Roaming\Python\Python310\Scripts
   C:\Users\your username\AppData\Roaming\Python\Python310\site-packages
   
    *Don't forget to replace - your username

6. Click **OK** and restart your PC
7. Check if the pipenv is installed correctly: 
    Command:- ***pipenv -h***
8. If it shows the following: 
   _Usage: pipenv [OPTIONS] COMMAND [ARGS].._
   Then, you’ve successfully installed the pipenv tool. Congrats!

Now, you can follow the commands mentioned in the documentation for further installation: https://docs.atrilabs.com/getting-started/installation/install_with_pipenv

Please feel free to ask if you face any issues during the installation.

**Comment by darshitac11** [comment]
This is a great writeup, thanks Chirag! 

In the upcoming version, we are making Python virtual environments optional to avoid these issues.

**Reply by chiragmathurid** [reply]
That's great to hear! I used your platform. It is truly an amazing. I would like to give my view that you can consider adding a few 'Templates' as an option to facilitate the process of starting and building web applications more quickly and efficiently. And please let me know if there is any way I can be of help. I would love to contribute to your project.

**Comment by cruxcode** [comment]
@chiragmathur03 are you in our Slack community? If not, it would be our pleasure to have you in our Slack community. Please join using the link https://join.slack.com/t/atricommunity/shared_invite/zt-1otyguhy8-P4OQQRWFWcMUsKk7yp0FHQ

**Reply by chiragmathurid** [reply]
Sure! Thank you for being so kind and inviting me.
